src/libkomwm.py

```diff
@@ -746,18 +746,18 @@ def rule_sort_key(dict_):
                             store_visibility(cl, 'line', st.get('object-id'), zoom)
                             dr_element.lines.extend([dr_line])
 
-                        if st.get('shield-font-size'):
-                            dr_element.shield.height = int(st.get('shield-font-size', 10))
-                            dr_element.shield.text_color = mwm_encode_color(colors, st, "shield-text")
-                            if st.get('shield-text-halo-radius', 0) != 0:
-                                dr_element.shield.text_stroke_color = mwm_encode_color(colors, st, "shield-text-halo", "white")
-                            dr_element.shield.color = mwm_encode_color(colors, st, "shield")
-                            if st.get('shield-outline-radius', 0) != 0:
-                                dr_element.shield.stroke_color = mwm_encode_color(colors, st, "shield-outline", "white")
-                            dr_element.shield.priority = get_drape_priority(cl, 'shield', st.get('object-id'))
-                            store_visibility(cl, 'shield', st.get('object-id'), zoom)
-                            if st.get('shield-min-distance', 0) != 0:
-                                dr_element.shield.min_distance = int(st.get('shield-min-distance', 0))
+                    if st.get('shield-font-size'):
+                        dr_element.shield.height = int(st.get('shield-font-size', 10))
+                        dr_element.shield.text_color = mwm_encode_color(colors, st, "shield-text")
+                        if st.get('shield-text-halo-radius', 0) != 0:
+                            dr_element.shield.text_stroke_color = mwm_encode_color(colors, st, "shield-text-halo", "white")
+                        dr_element.shield.color = mwm_encode_color(colors, st, "shield")
+                        if st.get('shield-outline-radius', 0) != 0:
+                            dr_element.shield.stroke_color = mwm_encode_color(colors, st, "shield-outline", "white")
+                        dr_element.shield.priority = get_drape_priority(cl, 'shield', st.get('object-id'))
+                        store_visibility(cl, 'shield', st.get('object-id'), zoom)
+                        if st.get('shield-min-distance', 0) != 0:
+                            dr_element.shield.min_distance = int(st.get('shield-min-distance', 0))
```

Comment:
Could you clarify: before, shield's style was under has_lines block, and now is out of has_lines scope. Why?

Reply:
It made shields dependent on presence of line drules in the same ::object_id.
I moved all shields from the default object id to ::shield to make them priority-independent from the default object id which contains line drules. So this "has_lines" condition prevented shield drules from ::shields objects to be processed.